unable to compile with 1.86.0

I'll admit i have very limited experience with rust

```
cargo version
cargo 1.86.0 (adf9b6ad1 2025-02-28)
```

```
cargo install tuisky
   Compiling tuisky v0.2.0
error[E0432]: unresolved import `bsky_sdk::api::agent::Session`
  --> /home/halkeye/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/tuisky-0.2.0/src/components/column.rs:11:5
   |
11 | use bsky_sdk::api::agent::Session;
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ no `Session` in `agent`

error[E0432]: unresolved import `bsky_sdk::api::agent::Session`
 --> /home/halkeye/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/tuisky-0.2.0/src/components/views/post.rs:5:5
  |
5 | use bsky_sdk::api::agent::Session;
  |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ no `Session` in `agent`

error[E0308]: mismatched types
   --> /home/halkeye/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/tuisky-0.2.0/src/components/modals/embed_record.rs:154:21
    |
154 |                     rkey,
    |                     ^^^^ expected `RecordKey`, found `String`

Some errors have detailed explanations: E0308, E0432.
For more information about an error, try `rustc --explain E0308`.
error: could not compile `tuisky` (lib) due to 3 previous errors
warning: build failed, waiting for other jobs to finish...
    Building [=======================> ] 300/302: bsky-sdk
error: failed to compile `tuisky v0.2.0`, intermediate artifacts can be found at `/tmp/cargo-installsbcQr1`.
To reuse those artifacts with a future compilation, set the environment variable `CARGO_TARGET_DIR` to that path.
```

--- a/Cargo.toml
+++ b/Cargo.toml
@@ -12,7 +12,7 @@ keywords = ["tui", "atproto", "bluesky", "atrium"]
 exclude = ["/config"]
 
 [dependencies]
-bsky-sdk = "0.1.12"
+bsky-sdk = "0.1.18"
 chrono = { version = "0.4.38", default-features = false }
 clap = { version = "4.5.8", features = ["derive"] }
 color-eyre = "0.6.3"

--- a/src/components/column.rs
+++ b/src/components/column.rs
@@ -8,7 +8,7 @@ use crate::backend::Watcher;
 use crate::config::Config;
 use crate::types::{Action, IdType};
 use bsky_sdk::agent::config::Config as AgentConfig;
-use bsky_sdk::api::agent::Session;
+use bsky_sdk::api::agent::atp_agent::AtpSession;
 use bsky_sdk::BskyAgent;
 use color_eyre::{eyre, Result};
 use crossterm::event::KeyEvent;
@@ -29,7 +29,7 @@ pub struct ColumnComponent {
     config: Config,
     action_tx: UnboundedSender<Action>,
     view_tx: UnboundedSender<ViewAction>,
-    session: Arc<RwLock<Option<Session>>>,
+    session: Arc<RwLock<Option<AtpSession>>>,
 }
 
 impl ColumnComponent {

--- a/src/components/modals/embed_record.rs
+++ b/src/components/modals/embed_record.rs
@@ -3,7 +3,7 @@ use super::types::{Action, Data};
 use super::ModalComponent;
 use bsky_sdk::agent::config::Config;
 use bsky_sdk::api::com::atproto::repo::strong_ref;
-use bsky_sdk::api::types::string::{AtIdentifier, Cid, Nsid};
+use bsky_sdk::api::types::string::{AtIdentifier, Cid, Nsid, RecordKey};
 use bsky_sdk::BskyAgent;
 use color_eyre::Result;
 use crossterm::event::{KeyCode, KeyEvent, KeyModifiers};
@@ -98,13 +98,13 @@ impl EmbedRecordModalComponent {
         ) else {
             return Some("missing authority, collection, or rkey");
         };
-        let (Ok(repo), Ok(collection)) = (
+        let (Ok(repo), Ok(collection), Ok(rkey)) = (
             authority.as_str().parse::<AtIdentifier>(),
             collection.as_str().parse::<Nsid>(),
+            rkey.as_str().parse::<RecordKey>(),
         ) else {
             return Some("invalid authority or collection");
         };
-        let rkey = rkey.as_str().to_string();
         let action_tx = self.action_tx.clone();
         let state = self.state.clone();
         tokio::spawn(async move {
@@ -133,7 +133,7 @@ impl EmbedRecordModalComponent {
             }
         }
     }
-    async fn try_get_record(collection: Nsid, repo: AtIdentifier, rkey: String) -> Result<Cid> {
+    async fn try_get_record(collection: Nsid, repo: AtIdentifier, rkey: RecordKey) -> Result<Cid> {
         let agent = BskyAgent::builder()
             .config(Config {
                 endpoint: PUBLIC_API_ENDPOINT.to_string(),

--- a/src/components/views/post.rs
+++ b/src/components/views/post.rs
@@ -2,7 +2,7 @@ use super::types::{Action, Data, Transition, View};
 use super::utils::{counts, profile_name, profile_name_as_str};
 use super::ViewComponent;
 use crate::backend::{Watch, Watcher};
-use bsky_sdk::api::agent::Session;
+use bsky_sdk::api::agent::atp_agent::AtpSession;
 use bsky_sdk::api::app::bsky::actor::defs::ProfileViewBasic;
 use bsky_sdk::api::app::bsky::embed::record::{self, ViewRecordRefs};
 use bsky_sdk::api::app::bsky::embed::record_with_media::ViewMediaRefs;
@@ -77,7 +77,7 @@ pub struct PostViewComponent {
     agent: Arc<BskyAgent>,
     watcher: Box<dyn Watch<Output = Union<OutputThreadRefs>>>,
     quit: Option<oneshot::Sender<()>>,
-    session: Option<Session>,
+    session: Option<AtpSession>,
 }
 
 impl PostViewComponent {
@@ -86,7 +86,7 @@ impl PostViewComponent {
         watcher: Arc<Watcher>,
         post_view: PostView,
         reply: Option<PostView>,
-        session: Option<Session>,
+        session: Option<AtpSession>,
     ) -> Self {
         let actions = Self::post_view_actions(&post_view, &session);
         let agent = watcher.agent.clone();
@@ -104,7 +104,7 @@ impl PostViewComponent {
             session,
         }
     }
-    fn post_view_actions(post_view: &PostView, session: &Option<Session>) -> Vec<PostAction> {
+    fn post_view_actions(post_view: &PostView, session: &Option<AtpSession>) -> Vec<PostAction> {
         let mut liked = None;
         if let Some(viewer) = &post_view.viewer {
             liked = viewer.like.as_ref();

